JPN-576 Community to-do list

perjg opened the pull request:
@Wikia/spitfires @d34th4ck3r

perjg committed:
JPN-576 Community to-do list

Grunny commented on extensions/wikia/CommunityPage/models/CommunityPageSpecialCommunityTodoListModel.class.php:
```diff
@@ -0,0 +1,27 @@
+class CommunityPageSpecialCommunityTodoListModel {
+	public function getData() {
+		$title = Title::newFromText(
+			wfMessage( 'communitypage-todo-module-page-name' )->inContentLanguage()->plain(),
```
What do you think of just making use of `wfMessage` and parse the contents with an output format? Also, having a customisable page name for this functionality will be difficult as each permutation would need to be whitelisted so admins could edit it. For this purpose, I'd suggest just making `Community-to-do-list` the page name and not have another message control that (MediaWiki pages have always been in English anyway).

Then, this method could just be something like (with any other functionality you need that can be added):

``` php
$message = wfMessage( 'community-to-do-list' );
$data = [
    'haveContent' => false,
];

if ( !$message->isDisabled() ) {
    $data['haveContent'] = true;
    $data['data'] = $message->parse();
}

return $data;
```

perjg replied:
@Grunny Your suggested alternate code works well, but I am now missing the edit link which was:

`'editUrl' => $title->getFullURL( ['action' => 'edit'] ),`

I tried `Title::newFromText(` $message->plain(), NS_MEDIAWIKI );` to get the Title, but it just gives null.

Any suggestions on how to get the proper edit url in the context of your suggested code?

Grunny replied:
I'd just do something like:

``` php
$messageKey = 'community-to-do-list';
$message = wfMessage( $messageKey );
$title = Title::newFromText( $messageKey, NS_MEDIAWIKI );
$data = [
    'haveContent' => false,
    'editUrl' => $title->getFullURL( ['action' => 'edit'] ),
];

if ( !$message->isDisabled() ) {
    $data['haveContent'] = true;
    $data['data'] = $message->parse();
}

return $data;
```

The message key could also be a constant on the class if it might need to be checked outside this method. What do you think?

perjg committed:
JPN-576 Cleaned up code, small scss adjustment

perjg committed:
Merge branch 'dev' into JPN-576

perjg commented:
@grunny: I've updated the code with your suggestion for getData implementation, it works great.

(my mistake was that I tried to call Title::newFromText from a wfMessage rather than from a string).

Grunny commented on extensions/wikia/CommunityPage/models/CommunityPageSpecialCommunityTodoListModel.class.php:
```diff
@@ -0,0 +1,20 @@
+		$title = Title::newFromText( $messageKey, NS_MEDIAWIKI );
+		$data = [
+			'haveContent' => false,
+			'editUrl' => $title->getFullURL( ['action' => 'edit'] ),
```
Minor: Spaces needed within square brackets per conventions.

Grunny commented on extensions/wikia/CommunityPage/scripts/ext.communityPage.js:
```diff
@@ -251,6 +251,15 @@ require([
+		});
+
+		$('.community-page-todo-list-module-content').on('mousedown touchstart', 'a', function (event) {
+			handleClick(event, 'community-page-todo-list-content');
```
I'd consider unifying these two tracking events into one, similar to what we do elsewhere in this code and use either a label or class to determine the tracking label.

Grunny commented:
Minor comments, otherwise LGTM. :)

perjg committed:
JPN-576 Cleanup. Removed link from 'contact an admin' page

perjg committed:
JPN-576 Set max height for todo list

perjg commented:
@Grunny I changed the tracking implementation a bit, but I kept the two separate selectors as I need to treat the data tag for the "edit" link and the actual todo list differently.

I also made a couple of other changes:
- Removed the link for "contact an admin" - it was based on my misunderstanding of the AC, there should be no link
- Added max-height 500px with scrollbar for todo list content.

Grunny commented on extensions/wikia/CommunityPage/scripts/ext.communityPage.js:
```diff
@@ -251,6 +256,18 @@ require([
+		$('.community-page-todo-list-module-content').on('mousedown touchstart', 'a', function (event) {
+			track({
+				category: 'community-page-todo-list-module',
+				label: event.delegateTarget.getAttribute('data-tracking'),
```
Ah, I see, since that's the case, I probably wouldn't bother with using the HTML data attribute for this label, as you're not going to use more than one, right?

perjg replied:
Yeah, the AC specifically say to use the same label for all content inside the .community-page-todo-list-module-content div, so I will drop the data-tracking attribute for that one and just hardcode the data to send.

Grunny commented on extensions/wikia/CommunityPage/scripts/ext.communityPage.js:
```diff
@@ -220,6 +220,11 @@ require([
 				label: label,
 			});
+
+			console.log({
```
Test code.

perjg committed:
JPN-576 Removed debug code

perjg committed:
JPN-576 Removed redundant data-tracking label

Grunny commented:
:+1:

perjg committed:
JPN-576 Minor CSS adjustment. Make edit link show up in new tab

perjg committed:
JPN-576 Downloaded new translations

perjg committed:
Merge branch 'dev' into JPN-576

Grunny commented:
:+1:

perjg committed:
Merge branch 'dev' into JPN-576

perjg committed:
JPN-576 Add padding

perjg committed:
Merge branch 'dev' into JPN-576

perjg committed:
JPN-576 Added latest translations

perjg committed:
JPN-576 Copy Change

perjg committed:
Merge branch 'dev' into JPN-576

perjg committed:
JPN-576 Show edit link for anybody with editinterface rights

perjg commented:
@Grunny A few more changes pushed to account for changes to the Acceptance Criteria.

Grunny commented:
LGTM.

perjg committed:
Merge branch 'dev' into JPN-576

perjg committed:
JPN-576 Download latest translations

perjg committed:
Merge branch 'dev' into JPN-576
